pkg/runtime/scheme/scheme.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheme
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// NewAddToScheme returns a function to add go struct types as GroupVersionKinds to a runtime.Scheme.
+// Scheme is used to map go structs to GroupVersionKinds and vice versa.
+func NewAddToScheme(gv schema.GroupVersion, types func() []runtime.Object) func(*runtime.Scheme) error {
```

Comment:
This is a really weird name, and seems like a weird return type. Find a better name, maybe?

Reply:
It creates a new `AddToScheme` function.